src/user/UserRouter.ts

```diff
@@ -1,20 +1,46 @@
-import { Request, Response, Router } from 'express';
+import { Request, Response, NextFunction, Router } from 'express';
 import UserService from './UserService';
 
 const router = Router();
 
-router.post('/api/1.0/users', async (req: Request, res: Response) => {
+const validateUsername = (req: Request, res: Response, next: NextFunction) => {
+  //
+
   const requestBody = req.body as { password: string; username: string; email: string };
 
   const { username } = requestBody;
 
   if (username === null) {
-    return res.status(400).send({
-      validationErrors: {
-        username: 'Username cannot be null',
-      },
-    });
+    req.validationErrors = {
+      ...req.validationErrors,
+      username: 'Username cannot be null',
+    };
+  }
+
+  next();
+};
+
+const validateEmail = (req: Request, res: Response, next: NextFunction) => {
+  const requestBody = req.body as { username?: string | null; password?: string | null; email?: string | null };
+
+  if (requestBody.email === null) {
+    req.validationErrors = {
+      ...req.validationErrors,
+      email: 'E-mail cannot be null',
+    };
   }
+  next();
+};
+
+router.post('/api/1.0/users', validateUsername, validateEmail, async (req: Request, res: Response) => {
+  // will take care of validation errors
+  if (req.validationErrors) {
```

Comment:
decided what to do if there are validationErrors and give a response 400 to user and the validationErrors fields